src/models/shape_loader.cpp

```diff
@@ -517,68 +916,124 @@ geo::ShapePtr loadShape(const std::string& model_path, tue::config::Reader cfg,
+            createPolygon(*shape, points, height, error, true);
         }
 
         cfg.endGroup();
     }
-    else if (cfg.readArray("compound") || cfg.readArray("group"))
+    else if (cfg.readArray("polyline")) // SDF ONLY
     {
-        boost::shared_ptr<geo::CompositeShape> composite(new geo::CompositeShape);
+        std::vector<geo::Vec2> points;
+        double height = 0.0;
         while(cfg.nextArrayItem())
         {
-            std::map<std::string, geo::ShapePtr> dummy_shape_cache;
-            geo::ShapePtr sub_shape = loadShape(model_path, cfg, dummy_shape_cache, error);
-            composite->addShape(*sub_shape, geo::Pose3D::identity());
+            std::string point_string;
+            if (cfg.value("point", point_string))
+            {
+                points.push_back(geo::Vec2());
+                geo::Vec2& p = points.back();
+                std::vector<std::string> point_vector = split(point_string, ' ');
+                p.x = std::stod(point_vector[0]);
+                p.y = std::stod(point_vector[1]);
+            }
+            else
+                cfg.value("height", height);
         }
         cfg.endArray();
+        if (height > 0 && std::abs<double>(height) >= 0.01)
+        {
+            shape.reset(new geo::Shape());
+            createPolygon(*shape, points, height, error, true);
+        }
+    }
+    else if (cfg.readGroup("mesh")) // SDF ONLY
+    {
+        std::string uri_path;
+        if (cfg.value("uri", uri_path))
+        {
+            tue::filesystem::Path mesh_path = getUriPath(uri_path);
+            if (mesh_path.exists())
+                shape = geo::Importer::readMeshFile(mesh_path.string());
+            else
+                error << "[ED::MODELS::LOADSHAPE] Mesh File: '" << mesh_path.string() << "' doesn't exist." << std::endl;
+        }
+        else
+            error << "[ED::MODELS::LOADSHAPE] No uri found for mesh." << std::endl;
+
+        std::string dummy;
+        if (cfg.value("submesh", dummy))
+            error << "[ED::MODELS::LOADSHAPE] 'submesh' of mesh is not supported by ED " << std::endl;
 
-        shape = composite;
     }
-    else if (cfg.readArray("heightmap"))
+    else if (cfg.readGroup("heightmap")) // SDF AND ED YAML
     {
         std::string image_filename;
         double height, resolution;
 
-        if (cfg.value("image", image_filename) && !image_filename.empty()
-                && cfg.value("resolution", resolution)
-                && cfg.value("height", height))
+        geo::Vec3 size;
+        if ((cfg.value("image", image_filename) && !image_filename.empty() && cfg.value("resolution", resolution) &&
+             cfg.value("height", height))) // ED YAML ONLY
         {
             std::string image_filename_full = image_filename;
 //            if (image_filename[0] == '/')
 //                image_filename_full = image_filename;
 //            else
 //                image_filename_full = model_path + "/" + image_filename;
```

Comment:
Remove comments

Reply:
I would like to keep it, as it was already there.